modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -554,9 +627,28 @@ export class Wallets implements IWallets {
     common.validateParams(params, ['walletShareId'], ['overrideEncryptedPrv', 'userPassword', 'newWalletPassphrase']);
 
     let encryptedPrv = params.overrideEncryptedPrv;
-
     const walletShare = (await this.getShare({ walletShareId: params.walletShareId })) as any;
-
+    if (walletShare.keychainOverrideRequired && walletShare.permissions.indexOf('admin') !== -1) {
+      if (_.isUndefined(params.userPassword)) {
+        throw new Error('userPassword param must be provided to decrypt shared key');
+      }
+      const walletKeychain = await this.createKeychain(params.userPassword);
```

Comment:
Why are we creating a new keychain? Shouldn't it just be an update?

Reply:
We only need to generate a new keychain when keychainOverrideRequired is set to true, indicating the intentional need to create a new keychain and replace the existing one in the wallet.